sympy/stats/error_prop.py

```diff
@@ -0,0 +1,44 @@
+"""Tools for arithmetic error propogation."""
+from itertools import repeat
+from sympy import Symbol, symbols, Add, Sum, Mul, simplify, Pow, exp
+
+def variance_prop(expr, consts=()):
+    """Symbolically propagates variance (sigma^2) for expressions.
+
+    Parameters
+    ==========
+    expr : Expr
+        A sympy expression to compute the variance for.
+    consts : sequence of Symbols, optional
+        Represents symbols that are known constants in the expr,
+        and thus have zero variance.
+
+    Returns
+    =======
+    var_expr : Expr
+        An expression for the total variance of the expr.
+        The variance for the original symbols (e.g. x) have a 'var_'
+        prepended to their name ('var_x').
+    """
+    args = expr.args
```

Comment:
> @leosartaj - there is a warning in the docs saying not to do this and 

By not including this the function does fail for even trivial inputs like, ``2``.

```
In [1]: from sympy.stats.error_prop import variance_prop

In [2]: variance_prop(2)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-941ca10acb95> in <module>()
----> 1 variance_prop(2)

/home/leosartaj/projects/sympy/sympy/stats/error_prop.pyc in variance_prop(expr, consts)
     21         prepended to their name ('var_x').
     22     """
---> 23     args = expr.args
     24     if len(args) == 0:
     25         if isinstance(expr, Symbol) and expr not in consts:

AttributeError: 'int' object has no attribute 'args'
```

Also, IMHO this is used abundantly in the present code. I feel that the sanitization should be done on the production side.

> it would be redundant on all the recursions.

For sympy objects ``sympify`` shouldn't take much computation. It simply returns without any changes.